lib/rexml/parsers/treeparser.rb

```diff
@@ -24,12 +24,20 @@ def parse
             #STDERR.puts "TREEPARSER GOT #{event.inspect}"
             case event[0]
             when :end_document
+              if in_doctype
+                raise ParseException.new("Malformed DOCTYPE: unclosed",
+                                         @parser.source, @parser)
+              end
```

Comment:
Can we do this in `baseparser.rb` instead of here?
If we can do it, other parsers such as `StreamParser` can detect this case.

Reply:
I will try it.

Reply:
Updated!